.flake8

```diff
@@ -0,0 +1,5 @@
+[flake8]
+ignore = E203, E266, E501, W503, F403, F401
```

Comment:
Where are those rules coming from? Also, in 2021, 88 char limit is a bit short IMO.

Reply:
I copied some of this config from this [blog post](https://ljvmiranda921.github.io/notebook/2018/06/21/precommits-using-black-and-flake8/). I did not look too closely yet at exactly why each of those are configured (and it's possible they're no longer needed - looks like the blog post is ~3 years old.

88 chars is the default length for black formatting, flake8 default is 79 chars.

Reply:
Here are the rules being ignored:

E203: Whitespace before ':'
E266: Too many leading '#' for block comment
E501: Line too long (82 > 79 characters) -- shouldn't need this if I'm specifying my own max length
W503: Line break occurred before a binary operator
F401: Module imported but unused
F403: 'from module import *' used; unable to detect undefined names

Reply:
More details in the [black docs](https://black.readthedocs.io/en/stable/the_black_code_style/current_style.html#line-length).

> If you’re using Flake8, you can bump max-line-length to 88 and mostly forget about it. However, it’s better if you use Bugbear’s B950 warning instead of E501, and bump the max line length to 88 (or the --line-length you used for black), which will align more with black’s “try to respect --line-length, but don’t become crazy if you can’t”. 

```
[flake8]
max-line-length = 88
...
select = C,E,F,W,B,B950
extend-ignore = E203, E501
```

Reply:
@ulgens I took the flake8 config from the black docs instead this time, as mentioned in my comment.